schema/map_test.go

```diff
@@ -344,3 +344,18 @@ func TestMapCompatibilityValidation(t *testing.T) {
 	assert.Error(t, s1.ValidateCompatibility(true))
 	assert.Error(t, s1.ValidateCompatibility([]string{}))
 }
+
+func TestMap_UnserializeIdempotent(t *testing.T) {
+	mapType := schema.NewTypedMapSchema[string, string](
+		schema.NewStringSchema(nil, nil, nil),
+		schema.NewStringSchema(nil, nil, nil),
+		nil,
+		schema.IntPointer(3),
```

Comment:
Min and Max on a Map type schema? Interesting. I glanced through some of the code, but didn't spend enough time to figure out all the paths and dependencies. So, does this mean that more than 3 pairs would fail validation?

Reply:
yes that is correct

Reply:
@jaredoconnell correct if me wrong

Reply:
I assume there are tests somewhere for *that*? I kinda wonder whether setting the min/max values here (or more specifically setting one but not the other) is loading more than necessary or perhaps supportable into these tests. (E.g., simply because it raises questions about the semantic context, which interferes with understanding a straightforward sequence ...)

Reply:
oh you mean, I don't need to have that argument in there. You're right. I can remove it from these tests. The min and max behavior is already tested.